ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecoderOptions.java

```diff
@@ -37,6 +38,34 @@
 
   void setRedisUri(ValueProvider<String> value);
 
+  @Description("If set to true, enable decryption of Pioneer payloads.")
+  @Default.Boolean(false)
+  Boolean getPioneerEnabled();
+
+  void setPioneerEnabled(Boolean value);
+
+  @Description("Path (local or gs://) to JSON array of metadata entries enumerating encrypted"
+      + " private keys, Cloud KMS resource ids for decrypting those keys, and their corresponding"
+      + " document namespaces; leave null to disable.")
+  ValueProvider<String> getPioneerMetadataLocation();
+
+  void setPioneerMetadataLocation(ValueProvider<String> value);
+
+  @Description("If set to true, assume that all private keys are encrypted with the associated"
+      + " KMS resourceId. Otherwise ignore KMS and assume all private keys are stored in plaintext."
+      + " This may be used for debugging.")
+  @Default.Boolean(true)
+  ValueProvider<Boolean> getPioneerKmsEnabled();
+
+  void setPioneerKmsEnabled(ValueProvider<Boolean> value);
+
+  @Hidden
+  @Description("Decompress pioneer pings.")
+  @Default.Boolean(true)
```

Comment:
Last I checked, default values were ignored for value providers in practice, although this is not documented. To provide a default value for a ValueProvider, you have to check at time of use and inject a default value there. We should remove this annotation as it has no effect.

Reply:
This does seem to work though, based on the main decoder test where I don't have to include the flag when launching the pipeline. I would like this value to always be true, but the decompression stage does require a `ValueProvider<boolean>` which needs to be statically defined (or provided before runtime).

Reply:
This is how `DecompressPayload` handles the value:

```
      if (enabled.isAccessible() && !enabled.get()) {
        // Compression has been explicitly turned off in options, so return unchanged message.
```

So `DecompressPayload` is essentially enforcing `true` as the default.

Reply:
Okay, that makes sense. My misunderstanding comes from the `kmsEnabled` option, where I've only tested the non-default value and assumed that defaults actually worked.

Reply:
I've verified the behavior of the `@Default.Boolean` annotation in a unit-test for `kmsEnabled`, let me know what your thoughts are here.

Reply:
It's certainly possible this has been addressed in a recent Beam release. It's also possible that this test only works because the option is provided at compile time. This may be a case where we get a different result if we publish the pipeline as a template and then only provide the option at runtime. We won't be able to know without testing. I'm fine going with this as written here and we'll see what happens when we start testing.

Reply:
It would be nice if this were fixed in a recent release, since this seems like it should be the correct behavior. Otherwise, perhaps a bug in beam? I haven't looked too closely at this to understand the technical limitations behind providing a default value to a value provider, so if it does need to be changed, it doesn't seem like it should be too large of a patch.